src/pages/home/sidebar/SidebarLinksData.js

```diff
@@ -0,0 +1,200 @@
+                isLoading={isLoading}
+                optionListItems={optionListItems}
+            />
+        </View>
+    );
+}
+
+SidebarLinksData.propTypes = propTypes;
+SidebarLinksData.defaultProps = defaultProps;
+SidebarLinksData.displayName = 'SidebarLinksData';
+
+/**
+ * This function (and the few below it), narrow down the data from Onyx to just the properties that we want to trigger a re-render of the component. This helps minimize re-rendering
+ * and makes the entire component more performant because it's not re-rendering when a bunch of properties change which aren't ever used in the UI.
+ * @param {Object} [report]
+ * @returns {Object|undefined}
+ */
+const chatReportSelector = (report) =>
+    report && {
+        reportID: report.reportID,
+        participants: report.participants,
+        participantAccountIDs: report.participantAccountIDs,
+        hasDraft: report.hasDraft,
+        isPinned: report.isPinned,
+        errorFields: {
+            addWorkspaceRoom: report.errorFields && report.errorFields.addWorkspaceRoom,
+        },
+        lastMessageText: report.lastMessageText,
+        lastVisibleActionCreated: report.lastVisibleActionCreated,
+        iouReportID: report.iouReportID,
+        total: report.total,
+        hasOutstandingIOU: report.hasOutstandingIOU,
+        statusNum: report.statusNum,
+        stateNum: report.stateNum,
+        chatType: report.chatType,
+        type: report.type,
+        policyID: report.policyID,
+        visibility: report.visibility,
+        lastReadTime: report.lastReadTime,
+        // Needed for name sorting:
+        reportName: report.reportName,
+        policyName: report.policyName,
+        oldPolicyName: report.oldPolicyName,
+        // Other less obvious properites considered for sorting:
+        ownerAccountID: report.ownerAccountID,
+        currency: report.currency,
+        managerID: report.managerID,
+        // Other important less obivous properties for filtering:
+        parentReportActionID: report.parentReportActionID,
+        parentReportID: report.parentReportID,
+    };
+
+/**
+ * @param {Object} [reportActions]
+ * @returns {Object|undefined}
+ */
+const reportActionsSelector = (reportActions) =>
+    reportActions &&
+    _.map(reportActions, (reportAction) => ({
+        errors: reportAction.errors,
+        message: [
+            {
+                moderationDecisions: [{decision: lodashGet(reportAction, 'message[0].moderationDecisions[0].decision')}],
+            },
+        ],
+    }));
+
+/**
+ * @param {Object} [policy]
+ * @returns {Object|undefined}
+ */
+const policySelector = (policy) =>
+    policy && {
+        type: policy.type,
+        name: policy.name,
+        avatar: policy.avatar,
+    };
+
+export default compose(
+    withCurrentReportID,
+    withNavigationFocus,
+    withOnyx({
+        chatReports: {
+            key: ONYXKEYS.COLLECTION.REPORT,
+            selector: chatReportSelector,
```

Comment:
This caused a "regression"? Without the use of the selector we used to pass the reports objects as is in Onyx. Then we mutated those objects here https://github.com/Expensify/App/blob/bba0eb2868fa95d483da1b380551bc46fe0df427/src/libs/SidebarUtils.js#L109-L110 by adding a `displayName` to each report. If you try to access `report.displayName` from anywhere in the App you'd be able to (because we modified the onyx data).

Now, we are using a selector and we are returning a copy of Onyx data, we are mutating a clone and not the onyx data. If we try to access `report.displayName` from anywhere outside the Sidebar we won't get a result which in turn caused this issue https://github.com/Expensify/App/issues/23348.

Reply:
Wow great catch! But at the same time I have to say that how it was working before is very dirty and we should find a different solution for it 😅

Reply:
I see that the proposal proposes a clean way, good job 💪